packages/react-native/src/theme/types.ts

```diff
@@ -0,0 +1,62 @@
+import { PartialDeep } from 'type-fest';
+import { ColorValue } from '@aws-amplify/ui';
+
+interface BaseTokens {
+  colors: Partial<Colors>;
+}
+
+export interface Tokens extends BaseTokens {
+  components: ComponentStyles;
+}
+
+type Override = Omit<StrictTheme, 'overrides'>;
+
+export type ColorMode = 'light' | 'dark' | 'system';
+
+/**
+ * A Theme just needs a name.
+ * Users can define any tokens they need, but they don't need a complete theme with all tokens.
+ */
+export interface Theme {
+  colorMode?: ColorMode;
+  /**
+   * The name of the theme.
+   */
+  name: string;
+  /**
+   * Component and component agnostic tokens.
+   */
+  tokens?: PartialDeep<Tokens>;
+  /**
+   * Overrides allow you to change design tokens in different contexts, like
+   * light and dark mode.
+   */
+  overrides?: Override[];
+}
+
+export interface ComponentStyles {
+  // TODO: add components
+}
+
+/**
+ * Fully built theme that has styling based
+ * on the design tokens and all design tokens have added fields
+ * to be used in Javascript/Typescript.
+ */
+export interface StrictTheme extends Theme {
+  colorMode: ColorMode;
+  tokens: Tokens;
+}
+
+// TODO: refactor everything below [color types and values] and use the output generated by style-dictionary with correct types
```

Comment:
We should just use the types we have defined in the UI package right? Right now a RN and RW theme are incompatible for the base tokens, and we are re-creating types and default values that already exist in a shared place.